File: src/test/java/pl/training/ForecastPage.java
package pl.training;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class ForecastPage {

    @FindBy(id = "pl.training.runkeeper:id/city_name_edit")
    public WebElement cityEdit;
    @FindBy(id ="pl.training.runkeeper:id/check_button")
    public WebElement checkButton;
    @FindBy(id = "pl.training.runkeeper:id/city_name_text")
    public WebElement city;

    public void enterCityName(String cityName) {
        cityEdit.sendKeys(cityName);
    }

    public void checkWeather() {
        checkButton.click();
    }

}

File: js/test/ForecastPage.js
class ForecastPage {

    constructor(driver) {
        this.driver = driver;
    }

    async cityEdit() {
        return await this.driver.$("id:pl.training.runkeeper:id/city_name_edit");
    }

    async checkButton() {
        return await this.driver.$("id:pl.training.runkeeper:id/check_button");
    }

    async city() {
        return await this.driver.$("id:pl.training.runkeeper:id/city_name_text");
    }

    async enterCityName(cityName) {
        const element = await this.cityEdit();
        await element.addValue(cityName);
    }

    async checkWeather() {
        const element = await this.checkButton()
        await element.click();
    }

}

module.exports = ForecastPage;